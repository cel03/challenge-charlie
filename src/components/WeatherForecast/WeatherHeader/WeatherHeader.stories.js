import React from 'react'
import { WeatherHeader } from '.'

export default {
  title: 'WeatherHeader',
  args: {
    cityName: 'Rio de Janeiro',
    stateName: 'Rio de Janeiro',
    onCityInputChange: () => {},
  },
}

const Template = (args) => <WeatherHeader {...args} />

export const Default = Template.bind({})
Default.storyName = 'Default'
