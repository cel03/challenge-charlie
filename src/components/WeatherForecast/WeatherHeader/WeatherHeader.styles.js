import styled from 'styled-components'
import { Container } from '../../../UI'

export const WeatherHeaderContainer = styled(Container)`
  color: rgb(126, 126, 124);
  background-color: rgba(243, 238, 236, 0.92);
  height: initial;
`

export const WeatherHeaderIcon = styled.div`
  width: auto;
  margin-right: 8px;
`

export const WeatherHeaderText = styled.span`
  width: 100%;
  text-align: start;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`
