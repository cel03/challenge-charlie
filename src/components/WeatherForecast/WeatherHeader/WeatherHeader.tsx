import React from 'react'
import { RiCompassLine } from 'react-icons/ri'
import { IconContext } from 'react-icons'

import {
  WeatherHeaderContainer,
  WeatherHeaderIcon,
  WeatherHeaderText,
} from './WeatherHeader.styles'

interface WeatherHeaderProps {
  city: string
  state: string
}

const WeatherHeader = ({ city, state }: WeatherHeaderProps) => {
  return (
    <WeatherHeaderContainer>
      <WeatherHeaderIcon>
        <RiCompassLine />
      </WeatherHeaderIcon>
      <WeatherHeaderText>
        {city}, {state}
      </WeatherHeaderText>
    </WeatherHeaderContainer>
  )
}

const WeatherHeaderWrapper = (props: WeatherHeaderProps) => (
  <IconContext.Provider value={{ size: '35' }}>
    <WeatherHeader {...props} />
  </IconContext.Provider>
)
export default WeatherHeaderWrapper
