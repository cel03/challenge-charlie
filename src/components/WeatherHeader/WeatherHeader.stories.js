import React from 'react'
import { WeatherHeader } from './'

export default {
  title: 'WeatherHeader',
  args: {
    cityName: 'Rio de Janeiro',
    stateName: 'Rio de Janeiro',
  },
}

const Wrapper = ({ children }) => (
  <div style={{ maxWidth: 300 }}>{children}</div>
)
const Template = (args) => (
  <Wrapper>
    <WeatherHeader {...args} />
  </Wrapper>
)

export const Default = Template.bind({})
Default.storyName = 'Default'
