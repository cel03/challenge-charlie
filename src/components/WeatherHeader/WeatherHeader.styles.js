import styled from 'styled-components'

export const WeatherHeaderContainer = styled.div`
  font-family: 'Trebuchet MS';
  color: rgb(126, 126, 124);
  background-color: rgba(243, 238, 236, 0.5);
  display: flex;
  align-items: center;
  text-align: center;
`

export const WeatherHeaderIcon = styled.div`
  min-width: 50px;
`

export const WeatherHeaderText = styled.div`
  width: 100%;
  text-align: start;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`
