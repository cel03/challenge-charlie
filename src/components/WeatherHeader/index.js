export { default as WeatherHeader } from './WeatherHeader'
