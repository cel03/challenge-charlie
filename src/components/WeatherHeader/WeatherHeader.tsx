import React from 'react'
import { RiCompassLine } from 'react-icons/ri'
import { IconContext } from "react-icons";

import { WeatherHeaderContainer, WeatherHeaderIcon, WeatherHeaderText } from './WeatherHeader.styles'

interface WeatherHeaderProps {
    cityName: string,
    stateName: string
}

const WeatherHeader = ({ cityName, stateName }: WeatherHeaderProps) => {
    return (
        <WeatherHeaderContainer >
            <WeatherHeaderIcon>
                <RiCompassLine />
            </WeatherHeaderIcon>
            <WeatherHeaderText>
                {cityName}, {stateName}
            </WeatherHeaderText>
        </WeatherHeaderContainer>
    )
}

const WeatherHeaderWrapper = (props: WeatherHeaderProps) => (
    <IconContext.Provider value={{ size: '35' }}>
        <WeatherHeader {...props} />
    </IconContext.Provider >
)
export default WeatherHeaderWrapper