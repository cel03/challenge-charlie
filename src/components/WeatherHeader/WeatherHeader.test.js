import React from 'react'
import { render } from '@testing-library/react'
import WeatherHeader from './WeatherHeader'

describe('WeatherHeader', () => {
  test('matches snapshot', () => {
    const { conatiner } = render(
      <WeatherHeader cityName="Rio de Janeiro" stateName="Rio de Janeiro" />
    )
    expect(conatiner).toMatchSnapshot()
  })
})
