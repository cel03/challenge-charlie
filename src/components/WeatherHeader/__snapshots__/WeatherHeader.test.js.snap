// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherHeader matches snapshot 1`] = `undefined`;
